src/JsonSchema/Constraints/Constraint.php

```diff
@@ -20,81 +20,15 @@
  * @author Robert Schönthal <seroscho@googlemail.com>
  * @author Bruno Prieto Reis <bruno.p.reis@gmail.com>
  */
-abstract class Constraint implements ConstraintInterface
+abstract class Constraint extends BaseConstraint implements ConstraintInterface
 {
-    protected $errors = array();
     protected $inlineSchemaProperty = '$schema';
 
+    const CHECK_MODE_NONE =             0x00000000;
     const CHECK_MODE_NORMAL = 		0x00000001;
     const CHECK_MODE_TYPE_CAST = 	0x00000002;
-
-    /**
-     * @var Factory
-     */
-    protected $factory;
-
-    /**
-     * @param Factory $factory
-     */
-    public function __construct(Factory $factory = null)
-    {
-        $this->factory = $factory ? : new Factory();
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function addError(JsonPointer $path = null, $message, $constraint='', array $more=null)
-    {
-        $error = array(
-            'property' => $this->convertJsonPointerIntoPropertyPath($path ?: new JsonPointer('')),
-            'pointer' => ltrim(strval($path ?: new JsonPointer('')), '#'),
-            'message' => $message,
-            'constraint' => $constraint,
-        );
-
-        if (is_array($more) && count($more) > 0)
-        {
-            $error += $more;
-        }
-
-        $this->errors[] = $error;
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function addErrors(array $errors)
-    {
-        if ($errors) {
-            $this->errors = array_merge($this->errors, $errors);
-        }
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function getErrors()
-    {
-        return $this->errors;
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    public function isValid()
-    {
-        return !$this->getErrors();
-    }
-
-    /**
-     * Clears any reported errors.  Should be used between
-     * multiple validation checks.
-     */
-    public function reset()
-    {
-        $this->errors = array();
-    }
+    const CHECK_MODE_COERCE_TYPES =   	0x00000004;
+    const CHECK_MODE_APPLY_DEFAULTS =  	0x00000008;
```

Comment:
Sorted - looks like there were tabs in there that shouldn't have been, and I have vim configured to display tabs using a 4-char tabstop, so I didn't see them. Well spotted!